cluster-location-metadata/README.md

```diff
@@ -0,0 +1,343 @@
+---
+creation_date: 2025-01-30
+issues: []
+owners:
+- https://github.com/orgs/giantswarm/teams/team-honeybadger
+state: review
+summary: The document discusses the need for consistent location metadata for clusters, addressing customer communication and management needs. It proposes a unified location identifier metadata system on the cluster level, for cloud and on-premises clusters.
+---
+
+# Location metadata for clusters
+
+### Problem statement
+
+We have anecdotal evidence that customers think about our installations in terms of their physical location. They communicate in terms like these (paraphrased):
+
+- the Dublin management cluster
+- the dev cluster in Singapore
+- the China installation
+
+Some on-prem customers use location identifiers like city names in cluster descriptions.
+
+Relevant user stories in this context could be, to name two examples:
+
+- As a platform admin, I want to check the Kubernetes version of all clusters in a certain location.
+- As a service owner, I want to check the deployment status of certain apps in a certain location.
+
+With Cluster API on cloud providers we usually have location indicators in cluster resources, in various places. Each cloud provider uses their own identifier system, however they all make use of a string identifier. The proposed solution in this RFC would help simplify and unify location lookup for clusters by clients throughout cloud providers.
+
+In on-premises installations, no location information is currently available in Kubernetes resources. In metrics (PrometheusMimir) we annotate many series with a `region` label. However, in on-prem installations, the value for this label is always `onprem`. This RFC attempts to change that.
+
+### Key assumptions
+
+This RFC assumes that each cluster exists in one location only, and nodes of the same cluster are not distributed geographically.
```

Comment:
Did we not just talk about scale to the cloud in the engineering workshop @JosephSalisbury ?

Reply:
yeah, i'm not 100% sure this is going to be true forever

however, even if we do start with some sort of "scale-to-the-cloud" solution in future, i think it's still reasonable to consider the cluster as being housed in one place (i.e: the control plane would still be colocated somewhere)

Reply:
That's true. This just made me think that maybe we should show the node labels but that would bé for later I guess